pkgs/tools/filesystems/ceph/default.nix

```diff
@@ -201,35 +203,7 @@ let
       # this also forces us to pin an older `pyopenssl` because the current one
       # is not compatible with older `cryptography`, see:
       #     https://github.com/pyca/pyopenssl/blob/d9752e44127ba36041b045417af8a0bf16ec4f1e/CHANGELOG.rst#2320-2023-05-30
-      cryptography = super.cryptography.overridePythonAttrs (old: rec {
-        version = cryptographyOverrideVersion;
-
-        src = fetchPypi {
-          inherit (old) pname;
-          version = cryptographyOverrideVersion;
-          hash = "sha256-KAPy+LHpX2FEGZJsfm9V2CivxhTKXtYVQ4d65mjMNHI=";
-        };
-
-        cargoDeps = rustPlatform.fetchCargoTarball {
-          inherit src;
-          sourceRoot = let cargoRoot = "src/rust"; in "${old.pname}-${cryptographyOverrideVersion}/${cargoRoot}";
-          name = "${old.pname}-${cryptographyOverrideVersion}";
-          hash = "sha256-gFfDTc2QWBWHBCycVH1dYlCsWQMVcRZfOBIau+njtDU=";
-        };
-
-        # Not using the normal `(old.patches or []) ++` pattern here to use
-        # the overridden package's patches, because current nixpkgs's `cryptography`
-        # has patches that do not apply on this old version.
-        patches = [
-          # Fix https://nvd.nist.gov/vuln/detail/CVE-2023-49083 which has no upstream backport.
-          # See https://github.com/pyca/cryptography/commit/f09c261ca10a31fe41b1262306db7f8f1da0e48a#diff-f5134bf8f3cf0a5cc8601df55e50697acc866c603a38caff98802bd8e17976c5R1893
-          ./python-cryptography-Cherry-pick-fix-for-CVE-2023-49083-on-cryptography-40.patch
-        ];
-
-        # Tests would require overriding `cryptography-vectors`, which is not currently
-        # possible/desired, see: https://github.com/NixOS/nixpkgs/pull/281858#pullrequestreview-1841421866
-        doCheck = false;
-      });
+      cryptography = self.cryptography_40;
```

Comment:
Why not just call the package from here?

```
cryptography = self.callPackage ./cryptography.nix {};
```

And with my Python maintainer hat on: I really don't want an old cryptography package in the python package set. People will ignore the warnings and reuse it not matter what. Comments do not help and there are too many PRs touching python packages to gatekeep this properly.

Reply:
At least `meta.knownVulnerabilities` *has* to be set if this is in `python3Packages`.

Reply:
> At least `meta.knownVulnerabilities` _has_ to be set if this is in `python3Packages`.

@dotlambda Why? As per comments above, the two mentioned vulnerabilities are patched by this PR. Are there more?

> Why not just call the package from here?

@mweinelt I believe I already answered that above (https://github.com/NixOS/nixpkgs/pull/334286#issuecomment-2287668225).

That said, I found another solution, adding it to `passthru` and then using `nix-build -A ceph.python.pkgs.cryptography`.

Please take another look.

Reply:
We often set `knownVulnerabilities` for EOL security libraries that won’t get further fixes even in the absence of specific flaws, but if it’s only accessible through the `ceph` package that might not be important. The current version seems better to me and we could always hide away the package further later if necessary, so I think I’d be happy to merge with this approach. The left‐over `cryptography_40` definition would presumably break eval, however.

Reply:
> The left‐over `cryptography_40` definition would presumably break eval, however.

Fixed!